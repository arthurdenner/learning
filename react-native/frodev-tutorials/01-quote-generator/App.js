import React from 'react';
import { Button, StyleSheet, Text, View } from 'react-native';
import { Font } from 'expo';
import quotes from './assets/quotes/quotes';

export default class App extends React.Component {
  state = {
    activeQuoteIndex: 0,
    isFontsLoaded: false,
  };

  componentDidMount() {
    Font.loadAsync({
      CreteRound: require('./assets/fonts/CreteRound-Regular.ttf'),
      Elsie: require('./assets/fonts/Elsie-Regular.ttf'),
    }).then(() => this.setState({
      isFontsLoaded: true,
    }));
  }

  nextQuote = () => {
    const { activeQuoteIndex } = this.state;
    if (activeQuoteIndex < quotes.length - 2) {
      this.setState({
        activeQuoteIndex: activeQuoteIndex + 1,
      });
    } else {
      this.setState({
        activeQuoteIndex: 0,
      });
    }
  }

  render() {
    const { activeQuoteIndex, isFontsLoaded } = this.state;
    const activeQuote = quotes[activeQuoteIndex];

    return (
      <View style={styles.container}>
        <Text style={[styles.message, isFontsLoaded && { fontFamily: 'Elsie' }]}>
          {activeQuote.message}
        </Text>
        <Text style={[styles.author, isFontsLoaded && { fontFamily: 'CreteRound' }]}>
          {activeQuote.author}
        </Text>
        <View style={styles.button}>
          <Button
            title="Next quote"
            onPress={this.nextQuote}
          />
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
    paddingHorizontal: 30,
  },
  message: {
    fontSize: 24,
    marginBottom: 10,
  },
  author: {
    fontSize: 18,
  },
  button: {
    position: 'absolute',
    bottom: 40,
  }
});
