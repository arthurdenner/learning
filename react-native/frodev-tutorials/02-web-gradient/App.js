import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import GradientsList from './components/gradients-list';

const App =() => (
  <View style={[styles.container]}>
    <View style={[styles.header]}>
      <Text style={[styles.h1]}>
        Web Gradients
      </Text>
      <Text style={[styles.h2]}>
        Colors from webgradients.com
      </Text>
    </View>
    <GradientsList />
  </View>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    justifyContent: 'center',
    paddingHorizontal: 10,
  },
  header: {
    paddingVertical: 10,
    alignItems: 'center',
  },
  h1: {
    fontSize: 24,
  },
  h2: {
    fontSize: 18,
  }
});

export default App;
