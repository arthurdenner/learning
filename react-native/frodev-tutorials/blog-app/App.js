import React from 'react';
import { Button, View } from 'react-native';
import { StackNavigator } from 'react-navigation';
import Blogs from '~/screens/blogs';
import Login from '~/screens/login';
import NewBlog from '~/screens/new-blog';
import { SCREENS } from '~/constants';

export default StackNavigator({
  [SCREENS.LOGIN]: {
    screen: Login,
    navigationOptions: {
      title: SCREENS.LOGIN,
    },
  },
  [SCREENS.BLOGS]: {
    screen: Blogs,
    navigationOptions: ({ navigation: { navigate } }) => ({
      title: SCREENS.BLOGS,
      headerRight: (
        <View style={{ marginRight: 10 }}>
          <Button
            title="New blog"
            onPress={() => navigate(SCREENS.NEW_BLOG)}
          />
        </View>
      ),
    }),
  },
  [SCREENS.NEW_BLOG]: {
    screen: NewBlog,
    navigationOptions: {
      title: SCREENS.NEW_BLOG,
    },
  },
}, {
  initialRouteName: SCREENS.NEW_BLOG,
});
