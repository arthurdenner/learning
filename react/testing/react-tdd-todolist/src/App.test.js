import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

let component;

describe('app component', () => {
  const items = ['Learn react', 'rest', 'go out'];

  beforeEach(() => {
    component = shallow(<App />);
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
  });

  it('should match its empty snapshot', () => {
    const tree = renderer.create(<App />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should add an item based on the value in the state', () => {
    const preventDefault = jest.fn();

    component.setState({ items });
    component.find('form').simulate('submit', { preventDefault });

    expect(toJson(component)).toMatchSnapshot();
    expect(preventDefault).toBeCalled();
  });

  it('should pass a selected value to the onChange function', () => {
    component.find('input').simulate('change', {
      target: { value: 'Change function' },
    });

    expect(toJson(component)).toMatchSnapshot();
  });
});
