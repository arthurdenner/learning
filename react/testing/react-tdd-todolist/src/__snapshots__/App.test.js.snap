// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app component should add an item based on the value in the state 1`] = `
<div
  style={
    Object {
      "padding": "1em",
    }
  }
>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      value=""
    />
    <button
      style={
        Object {
          "marginLeft": "1em",
        }
      }
      type="submit"
    >
      Submit
    </button>
  </form>
  <List
    items={
      Array [
        "Learn react",
        "rest",
        "go out",
        "",
      ]
    }
  />
</div>
`;

exports[`app component should match its empty snapshot 1`] = `
<div
  style={
    Object {
      "padding": "1em",
    }
  }
>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      value=""
    />
    <button
      style={
        Object {
          "marginLeft": "1em",
        }
      }
      type="submit"
    >
      Submit
    </button>
  </form>
  <ul />
</div>
`;

exports[`app component should pass a selected value to the onChange function 1`] = `
<div
  style={
    Object {
      "padding": "1em",
    }
  }
>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      value="Change function"
    />
    <button
      style={
        Object {
          "marginLeft": "1em",
        }
      }
      type="submit"
    >
      Submit
    </button>
  </form>
  <List
    items={Array []}
  />
</div>
`;
